File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        answer = ListNode(0)
        answer_current = answer
        carry = 0

        while 1:
            l1_val = l1.val if l1 is not None else 0
            l2_val = l2.val if l2 is not None else 0

            sum_val = (l1_val + l2_val + carry)
            answer_current.val = sum_val%10
            carry = sum_val//10

            if l1:
                l1 = l1.next
            if l2:
                l2 = l2.next

            if l1 or l2 or carry:
                answer_current.next = ListNode(0)
                answer_current = answer_current.next
            else:
                break
            
        return answer
            

File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode answer = new ListNode(0);
        ListNode head = answer;
        int carry = 0;

        while (true){
            int l1Val = (l1 != null) ? l1.val : 0;
            int l2Val = (l2 != null) ? l2.val : 0;
            
            int sum = l1Val + l2Val + carry;
            int rest = sum % 10;
            carry = sum / 10;
            
            if (l1 != null) {l1 = l1.next;}
            if (l2 != null) {l2 = l2.next;}

            ListNode newNode = new ListNode(rest);
            answer.next = newNode;
            answer = newNode;

            if ((carry == 0) && ((l1 == null) && (l2==null))){
                break;
            }
        }

        return head.next;
    }
}